app/Http/Controllers/CategoryController.php

```diff
@@ -8,10 +8,15 @@
 
 class CategoryController extends Controller
 {
+    public function index()
+    {
+        return Category::all();
+    }
+
     public function store(Request $request)
     {
         $validator = Validator::make($request->all(), [
-            'name' => ['required', 'string', 'max:40', 'regex:/(^([a-zA-Z]+)(\d+)?$)/u'],
+            'name' => ['required', 'string', 'max:40', 'regex:/(^([a-zA-Z]+)(\d+)?$)/u', 'unique:categories'],
```

Comment:
In case of soft delete we might have errors, It that case you should build your own custom rule

Reply:
we will hard delete the categories, we will not face an error.